js/functions.js

```diff
@@ -89,6 +92,12 @@ async function fetchFirmwareDownloadPage() {
  */
 function FindDeviceRow(table, keyboardName) {
     // table -> tbody -> trs
+
+    if(keyboardName == null) {
```

Comment:
I think a check should be done on the conversion instead, to make sure we get a usable result, otherwise there is no point in checking for updates on null.

Reply:
Indeed! I added this check here is purely to work around keyboardName being undefined. maybe adding a function dedicated to looking up device name that returns Unknown keyboard or some placeholder that can be used for further code to check so it doesn't panic.